modules/sugar/src/test/ts/browser/FocusTest.ts

```diff
@@ -31,3 +37,96 @@ UnitTest.test('FocusTest', () => {
 
   Remove.remove(div);
 });
+
+UnitTest.test('Focus.active in ShadowRoot', () => {
+  withShadowElement((sr, id, sh) => {
+    const innerInput: Element<HTMLElement> = Element.fromTag('input');
+    Insert.append(sr, innerInput);
+
+    const outerInput: Element<HTMLElement> = Element.fromTag('input');
+    Insert.append(Body.body(), outerInput);
+
+    Focus.focus(innerInput);
+    Assert.eq('ShadowRoot\'s active element is the inner input box', Focus.active(sr), Option.some(innerInput), tOption(tElement()));
+    Assert.eq('Document\'s active element is the shadow host', Focus.active(), Option.some(sh), tOption(tElement()));
+
+    Focus.focus(outerInput);
+    Assert.eq('ShadowRoot\'s active element should be none', Focus.active(sr), Option.none(), tOption(tElement()));
+    Assert.eq('Document\'s active element is the outer input box', Focus.active(), Option.some(outerInput), tOption(tElement()));
+
+    Remove.remove(outerInput);
+  });
+});
+
+UnitTest.test('Focus.search in ShadowRoot', () => {
+  withShadowElement((sr, id, sh) => {
+    const innerInput: Element<HTMLElement> = Element.fromTag('input');
+    Insert.append(id, innerInput);
+
+    const outerInput: Element<HTMLElement> = Element.fromTag('input');
+    Insert.append(Body.body(), outerInput);
+
+    Focus.focus(innerInput);
+    Assert.eq('Searching from div inside shadow root should yield focused input box', Focus.search(id), Option.some(innerInput), tOption(tElement()));
+    Assert.eq('Searching from shadow root should yield focused input box', Focus.search(sr), Option.some(innerInput), tOption(tElement()));
+    Assert.eq('Searching from shadow host should yield shadow host', Focus.search(sh), Option.some(sh), tOption(tElement()));
+    Assert.eq('Searching from body should yield shadow host', Focus.search(Body.body()), Option.some(sh), tOption(tElement()));
+
+    Focus.focus(outerInput);
+    Assert.eq('Searching from div inside shadow root should yield none', Focus.search(id), Option.none(), tOption(tElement()));
+    Assert.eq('Searching from shadow root should yield none', Focus.search(sr), Option.none(), tOption(tElement()));
+    Assert.eq('Searching from shadow host should yield none', Focus.search(sh), Option.none(), tOption(tElement()));
+    Assert.eq('Searching from body should yield outer input box', Focus.search(Body.body()), Option.some(outerInput), tOption(tElement()));
+
+    Remove.remove(outerInput);
+  });
+});
+
+UnitTest.test('Focus.hasFocus in ShadowRoot', () => {
+  withShadowElement((shadowRoot, innerDiv, shadowHost) => {
+    const innerInput: Element<HTMLElement> = Element.fromTag('input');
+    Insert.append(innerDiv, innerInput);
+
+    const outerInput: Element<HTMLElement> = Element.fromTag('input');
```

Comment:
Should `outerInput` be removed at the end of the function like the other unit tests above?

Reply:
Yep. Nice catch.
Fixed.